nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -588,26 +588,184 @@ in {
+        locations."= /sw.js" = {
+          tryFiles = "$uri =404";
+          priority = 2100;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=604800, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
+        };
+
+        locations."^~ /assets/" = {
+          tryFiles = "$uri =404";
+          priority = 2210;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
+        };
+
+        locations."^~ /avatars/" = {
+          tryFiles = "$uri =404";
+          priority = 2220;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
+        };
+
+        locations."^~ /emoji/" = {
+          tryFiles = "$uri =404";
+          priority = 2230;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
+        };
+
+        locations."^~ /headers/" = {
+          tryFiles = "$uri =404";
+          priority = 2240;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
+        };
+
+        locations."^~ /packs/" = {
+          tryFiles = "$uri =404";
+          priority = 2250;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc 'h3=":443"; ma=86400';
+            ''}
+          '';
```

Comment:
If this is the same in multiple places we should put that onto a variable.

Reply:
Fixed!